src/prefect/orion/schemas/actions.py

```diff
@@ -77,6 +79,22 @@ class FlowUpdate(ActionBaseModel):
 class DeploymentCreate(ActionBaseModel):
     """Data used by the Orion API to create a deployment."""
 
+    @root_validator(pre=True)
+    def remove_worker_pool_queue_id(cls, values):
+        # 2.7.7 removed worker_pool_queue_id in lieu of worker_pool_name and
+        # worker_pool_queue_name. This validator removes worker_pool_queue_id
+        # to avoid 422 errors when it is provided by older clients.
+        values_copy = deepcopy(values)
+        worker_pool_queue_id = values_copy.pop("worker_pool_queue_id", None)
+        if worker_pool_queue_id:
+            warnings.warn(
```

Comment:
this feels like a lot to add for an experimental feature that no one should be using yet; that prompts my question: should we be only passing them through the client _if_ they are not `None` instead?

Reply:
That makes sense to me and increases client/server compatibility!

Reply:
That absolutely should be our approach in the future. Unfortunately this is necessary to mitigate for already released clients.